sda/cmd/api/api.go

```diff
@@ -534,41 +570,106 @@ func downloadFile(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+/*
+setAccession handles requests to assign an accession ID to a file.
+This endpoint supports two input modes:
+1. By query parameters ("fileid" and "accessionid"): Retrieves user, file path, and decrypted checksum from the database using the file ID.
+2. By JSON payload: Expects a JSON body with user and file path, then looks up the correlation ID and decrypted checksum.
+If both query parameters and a JSON payload are provided, the request is rejected with a 400 Bad Request.
+The function constructs an accession message, validates it and sends it to the message broker.
+*/
 func setAccession(c *gin.Context) {
-	var accession schema.IngestionAccession
-	if err := c.BindJSON(&accession); err != nil {
-		c.AbortWithStatusJSON(
-			http.StatusBadRequest,
-			gin.H{
-				"error":  "json decoding : " + err.Error(),
-				"status": http.StatusBadRequest,
-			},
-		)
+	var (
+		accession schema.IngestionAccession
+		corrID    string
+	)
+	hasQuery := c.Query("fileid") != "" || c.Query("accessionid") != ""
+	missingAccession := c.Query("fileid") != "" && c.Query("accessionid") == ""
+	hasBody := c.Request.ContentLength > 0
+	switch {
+	case hasQuery && hasBody:
+		c.AbortWithStatusJSON(http.StatusBadRequest, "both parameters and json payload provided. Choose one")
 
 		return
-	}
+	case missingAccession:
+		c.AbortWithStatusJSON(http.StatusBadRequest, "accessionid is not provided")
 
-	corrID, err := Conf.API.DB.GetCorrID(accession.User, accession.FilePath, "")
-	if err != nil {
-		if corrID == "" {
-			c.AbortWithStatusJSON(http.StatusBadRequest, err.Error())
-		} else {
-			c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	case c.Query("fileid") != "" && c.Query("accessionid") != "":
+		// Get the user and the inbox filepath
+		fileDetails, err := Conf.API.DB.GetFileDetailsFromUUID(c.Query("fileid"), "verified")
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, "file details not found")
+
+			return
 		}
+		// Get the decrypted checksum
+		fileDecrChecksum, err := Conf.API.DB.GetDecryptedChecksum(c.Query("fileid"))
+		if err != nil {
+			log.Debugln(err.Error())
+			c.AbortWithStatusJSON(http.StatusNotFound, "decrypted checksum not found")
```

Comment:
```suggestion
			c.AbortWithStatusJSON(http.StatusInternalServerError, "required data missing")
```

A 404 would be if the FILE-ID doesn't exist.